x-pack/plugins/data_visualizer/public/application/index_data_visualizer/locator/locator.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// @ts-ignore
+import { encode } from 'rison-node';
+import { stringify } from 'query-string';
+import { SerializableState } from '../../../../../../../src/plugins/kibana_utils/common';
+import { RefreshInterval, TimeRange } from '../../../../../../../src/plugins/data/common';
+import { LocatorDefinition, LocatorPublic } from '../../../../../../../src/plugins/share/common';
+import { QueryState } from '../../../../../../../src/plugins/data/public';
+import { Dictionary, isRisonSerializationRequired } from '../../common/util/url_state';
+import { SearchQueryLanguage } from '../types/combined_query';
+
+export const DATA_VISUALIZER_APP_LOCATOR = 'DATA_VISUALIZER_APP_LOCATOR';
+
+export interface IndexDataVisualizerLocatorParams extends SerializableState {
+  /**
+   * Optionally set saved search ID.
+   */
+  savedSearchId?: string;
+
+  /**
+   * Optionally set index pattern ID.
+   */
+  indexPatternId?: string;
+
+  /**
+   * Optionally set the time range in the time picker.
+   */
+  timeRange?: TimeRange;
+
+  /**
+   * Optionally set the refresh interval.
+   */
+  refreshInterval?: RefreshInterval & SerializableState;
+
+  /**
+   * Optionally set a query.
+   */
+  query?: {
+    searchQuery: SerializableState;
+    searchString: string | SerializableState;
+    searchQueryLanguage: SearchQueryLanguage;
+  };
+
+  /**
+   * If not given, will use the uiSettings configuration for `storeInSessionStorage`. useHash determines
+   * whether to hash the data in the url to avoid url length issues.
+   */
+  useHash?: boolean;
+  /**
+   * Optionally set visible field names.
+   */
+  visibleFieldNames?: string[];
+  /**
+   * Optionally set visible field types.
+   */
+  visibleFieldTypes?: string[];
+}
+
+export type IndexDataVisualizerLocator = LocatorPublic<IndexDataVisualizerLocatorParams>;
+
+export class IndexDataVisualizerLocatorDefinition
+  implements LocatorDefinition<IndexDataVisualizerLocatorParams> {
+  public readonly id = DATA_VISUALIZER_APP_LOCATOR;
+
+  constructor() {}
+
+  public readonly getLocation = async (params: IndexDataVisualizerLocatorParams) => {
```

Comment:
there are no capabilities, space or license checks performed here. if the user doesn't have permission to use ML or ML is disabled in the space or in elasticsearch then this link shouldn't be shown.